src/runtime/math.py

```diff
@@ -87,6 +86,167 @@ def ceil(x: f32) -> i32:
         return r
     return r + 1
 
+# fsum
+# supported data types: i8, i16, i32, i64, f32, f64
+
+@overload
+def fsum(n: i32, arr: i8[:]) -> f64:
+    """
+    Sum of the elements of `arr`.
+    """
+    sum: f64
+    sum = 0.0
+
+    i: i32
+    for i in range(n):
+        sum += arr[i]
+    return sum
+
+@overload
+def fsum(n: i32, arr: i16[:]) -> f64:
+    """
+    Sum of the elements of `arr`.
+    """
+    sum: f64
+    sum = 0.0
+
+    i: i32
+    for i in range(n):
+        sum += arr[i]
+    return sum
+
+@overload
+def fsum(n: i32, arr: i32[:]) -> f64:
+    """
+    Sum of the elements of `arr`.
+    """
+    sum: f64
+    sum = 0.0
+
+    i: i32
+    for i in range(n):
+        sum += arr[i]
+    return sum
+
+@overload
+def fsum(n: i32, arr: i64[:]) -> f64:
+    """
+    Sum of the elements of `arr`.
+    """
+    sum: f64
+    sum = 0.0
+
+    i: i32
+    for i in range(n):
+        sum += arr[i]
+    return sum
+
+@overload
+def fsum(n: i32, arr: f32[:]) -> f64:
+    """
+    Sum of the elements of `arr`.
+    """
+    sum: f64
```

Comment:
The API differs here: https://docs.python.org/3/library/math.html. We have `size` as the argument which is not available in CPython's `math.py`